torch/csrc/jit/script/module.h

```diff
@@ -47,10 +46,12 @@ struct Module;
 
 template <typename T>
 struct slot_list_impl;
-using slot_list = slot_list_impl<Slot>;
-using module_list = slot_list_impl<Module>;
+using module_list = slot_list_impl<std::pair<std::string, Module>>;
```

Comment:
My comment here got eaten somehow: this is a readability regression, as the fields `name` and `value` have replaced by `first` and `second`. I don't think it's a huge deal since `first` and `second` are STL convention, but it could be worth retaining a small struct for naming here.

Reply:
I agree, but do think this is a big regression in understandability of the API. I think use of std::pair and std::tuple are almost always wrong. We should at the very least have a struct with names and not replace a bunch of types with auto in the code.